x-pack/metricbeat/module/panw/system/resources.go

```diff
@@ -0,0 +1,324 @@
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/panw"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+const resourceQuery = "<show><system><resources></resources></system></show>"
+
+var resourcesLogger *logp.Logger
+
+func getResourceEvents(m *MetricSet) ([]mb.Event, error) {
+	// Set logger so all the parse functions have access
+	resourcesLogger = m.logger
+
+	var response ResourceResponse
+	output, err := m.client.Op(resourceQuery, panw.Vsys, nil, &response)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute operation: %w", err)
+	}
+
+	err = xml.Unmarshal(output, &response)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal XML response: %w", err)
+	}
+
+	events := formatResourceEvents(m, response.Result)
+
+	return events, nil
+}
+
+/*
+Output from the XML API call is the standard "top" output:
+
+top - 07:51:37 up 108 days,  1:38,  0 users,  load average: 5.52, 5.79, 5.99
+Tasks: 189 total,   7 running, 182 sleeping,   0 stopped,   0 zombie
+%Cpu(s): 73.0 us,  4.6 sy,  0.0 ni, 21.7 id,  0.0 wa,  0.0 hi,  0.7 si,  0.0 st
+MiB Mem :   5026.9 total,    414.2 free,   2541.5 used,   2071.1 buff/cache
+MiB Swap:   5961.0 total,   4403.5 free,   1557.6 used.   1530.0 avail Mem
+
+	 PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
+	5692       20   0  121504   8396   6644 R  94.4   0.2 155491:08 pan_task
+	5695       20   0  121504   7840   6632 R  94.4   0.2 155491:01 pan_task
+	5696       20   0  121504   8360   6816 R  94.4   0.2 155486:29 pan_task
+	5699       20   0  121504   8132   6676 R  94.4   0.2 155236:17 pan_task
+	5700       20   0  146304  18424   6780 R  88.9   0.4 155491:40 pan_task
+
+22360 nobody    20   0  459836  40592  10148 R  22.2   0.8   0:38.65 httpd
+
+	6374       17  -3 1078156  18272   9716 S   5.6   0.4 215:46.41 routed
+
+14227       20   0   18108   7184   2172 R   5.6   0.1   0:00.04 top
+
+	1       20   0    2532    696    656 S   0.0   0.0   3:48.14 init
+	2       20   0       0      0      0 S   0.0   0.0   0:00.83 kthreadd
+*/
+func formatResourceEvents(m *MetricSet, input string) []mb.Event {
+	timestamp := time.Now()
+	events := make([]mb.Event, 0)
+
+	// We only need the top 5 lines
+	lines := strings.Split(input, "\n")
+	lines = lines[:5]
+
+	systemInfo := parseSystemInfo(lines[0])
+	taskInfo := parseTaskInfo(lines[1])
+	cpuInfo := parseCPUInfo(lines[2])
+	memoryInfo := parseMemoryInfo(lines[3])
+	swapInfo := parseSwapInfo(lines[4])
+
+	event := mb.Event{
+		Timestamp: timestamp,
+		MetricSetFields: mapstr.M{
+			"uptime": mapstr.M{
+				"days":    systemInfo.Uptime.Days,
+				"hours":   systemInfo.Uptime.Hours,
+				"minutes": systemInfo.Uptime.Minutes,
+			},
+			"user_count": systemInfo.UserCount,
+			"load_average": mapstr.M{
+				"1m":  systemInfo.LoadAverage.OneMinute,
+				"5m":  systemInfo.LoadAverage.FiveMinute,
+				"15m": systemInfo.LoadAverage.FifteenMinute,
```

Comment:
Im not sure if it's ok for a metric to start with a digit, seems strange. Can we use something like `one_minute` etc maybe?